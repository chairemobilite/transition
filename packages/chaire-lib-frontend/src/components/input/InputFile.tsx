/*
 * Copyright 2022, Polytechnique Montreal and contributors
 *
 * This file is licensed under the MIT License.
 * License text available at https://opensource.org/licenses/MIT
 */
import React from 'react';

export type InputFileProps = {
    id: string;
    onChange?: React.ChangeEventHandler<HTMLInputElement>;
    accept?: string;
    inputRef?: React.RefObject<HTMLInputElement>;
    disabled?: boolean;
};

const InputFile: React.FunctionComponent<InputFileProps> = ({
    id,
    onChange,
    accept,
    inputRef,
    disabled
}: InputFileProps) => {
    const inputAttributes = {
        autoComplete: 'none',
        type: 'file',
        accept: accept,
        className: 'tr__form-input-file apptr__input _input-file',
        name: id,
        id: id,
        onChange: onChange,
        disabled: disabled === true ? true : undefined
    };

    return <input {...inputAttributes} ref={inputRef} />;
};

export default InputFile;
