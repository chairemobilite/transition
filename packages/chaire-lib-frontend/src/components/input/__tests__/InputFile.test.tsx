/*
 * Copyright 2022, Polytechnique Montreal and contributors
 *
 * This file is licensed under the MIT License.
 * License text available at https://opensource.org/licenses/MIT
 */
import * as React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';

import InputFile from '../InputFile';

const mockOnChange = jest.fn();
const testId = 'fileWidgetId';
const testLabel = 'Upload file';

test('Default props', () => {
    const { container } = render(<InputFile
        id = {testId}
    />);
    expect(container).toMatchSnapshot();
});

test('All props', () => {
    const ref: React.RefObject<HTMLInputElement> = React.createRef() as React.RefObject<HTMLInputElement>;
    const { container } = render(<InputFile
        id = {testId}
        onChange = {mockOnChange}
        accept = "*.zip"
        inputRef = {ref}
        disabled = {false}
    />);
    expect(container).toMatchSnapshot();
});

test('Disabled', () => {
    const ref: React.RefObject<HTMLInputElement> = React.createRef() as React.RefObject<HTMLInputElement>;
    const { container } = render(<InputFile
        id = {testId}
        onChange = {mockOnChange}
        accept = "*.zip"
        inputRef = {ref}
        disabled = {true}
    />);
    expect(container).toMatchSnapshot();
});

test('Call onChange', () => {
    mockOnChange.mockClear();
    const { getByLabelText } = render(
        <div>
            <InputFile
                id = {testId}
                onChange = {mockOnChange}
                accept = "*.zip"
            />
            <label htmlFor={testId}>{testLabel}</label>
        </div>);
    const input = getByLabelText(testLabel);
    fireEvent.change(input, { target: { value: '' } });
    expect(mockOnChange).toHaveBeenCalledTimes(1);
});
