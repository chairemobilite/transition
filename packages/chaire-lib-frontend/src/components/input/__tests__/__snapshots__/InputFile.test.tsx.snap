// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All props 1`] = `
<div>
  <input
    accept="*.zip"
    autocomplete="none"
    class="tr__form-input-file apptr__input _input-file"
    id="fileWidgetId"
    name="fileWidgetId"
    type="file"
  />
</div>
`;

exports[`Default props 1`] = `
<div>
  <input
    autocomplete="none"
    class="tr__form-input-file apptr__input _input-file"
    id="fileWidgetId"
    name="fileWidgetId"
    type="file"
  />
</div>
`;

exports[`Disabled 1`] = `
<div>
  <input
    accept="*.zip"
    autocomplete="none"
    class="tr__form-input-file apptr__input _input-file"
    disabled=""
    id="fileWidgetId"
    name="fileWidgetId"
    type="file"
  />
</div>
`;
